src/loki/search.cc

```diff
@@ -574,13 +493,30 @@ struct bin_handler_t {
       }
 
       // if we already have a better reachable candidate we can just assume this one is reachable
-      auto reachability = check_reachability(begin, end, tile, edge);
+      auto reach = check_reachability(begin, end, tile, edge);
 
       // keep the best point along this edge if it makes sense
       c_itr = bin_candidates.begin();
       for (p_itr = begin; p_itr != end; ++p_itr, ++c_itr) {
-        // which batch of findings
-        bool reachable = reachability >= p_itr->location.minimum_reachability_;
+        // is this edge reachable in the right way
+        bool reachable = reach.outbound >= p_itr->location.min_outbound_reach_ &&
+                         reach.inbound >= p_itr->location.min_inbound_reach_;
+        // it's possible that it isnt reachable but the opposing is, switch to that if so
+        const GraphTile* opp_tile = tile;
+        const DirectedEdge* opp_edge = nullptr;
+        if (!reachable && (opp_edge = reader.GetOpposingEdge(edge, opp_tile)) &&
+            edge_filter(opp_edge) > 0.f) {
+          auto opp_reach = check_reachability(begin, end, opp_tile, opp_edge);
+          if (opp_reach.outbound >= p_itr->location.min_outbound_reach_ &&
+              opp_reach.inbound >= p_itr->location.min_inbound_reach_) {
+            tile = opp_tile;
+            edge = opp_edge;
+            reach = opp_reach;
+            reachable = true;
+          }
+        }
```

Comment:
this is an important little bit of  code. for regular routing, at the origin you need only outbound reach and at the destination you need only inbound reach (intermediate points need both). loki only checks one directed edge per normal edge in the graph. this meant that it wasnt checking reachability of the opposing edge. there are some cases were the best edge for an origin for example would deadend into a gate, but the opposing edge, going away from the gate would have fine reachability. this check allows us, in the case that the other edge didnt make the cut, to check if the opposing edge does.

you might notice that the check for reachable is per input location `p_iter` and that means this can flip each time. that is ok because the `check_reachability` function caches the result so it doesnt actually do more than 2 reach checks here.